src/notebooksyncagent.cpp

```diff
@@ -750,6 +751,7 @@ bool NotebookSyncAgent::calculateDelta(
     // and createdBefore; so instead we have to build a datetime which "should" satisfy
     // the inequality for all possible local modifications detectable since the last sync.
     KDateTime syncDateTime = mNotebook->syncDate().addSecs(1); // deleted after, created before...
+    KDateTime earliestAnchorTimestamp = syncDateTime < fromDate ? syncDateTime : fromDate;
```

Comment:
Hrm... Actually... I'm not sure about this after all...

mkcal uses an inequality: SELECT * from Components WHERE created < fromDate AND modified > fromDate;

Similarly for deletions: WHERE created < fromDate AND deleted > fromDate;

It's possible that we NEED to do the unite in order to catch both the ones which were created before the earlier of the two timestamps, AS WELL as the ones which were created AFTER the earlier timestamp but BEFORE the later timestamp...